src/config/configFile/dataDirRepository.js

```diff
@@ -0,0 +1,26 @@
+const fs = require('fs');
+const path = require('path');
+const os = require('os');
+
+const CouldNotCreateDatadirError = require('../errors/CouldNotCreateDatadirError');
+
+class DataDirRepository {
```

Comment:
I would call it home dir. But nor sure that repository pattern fits here. A repository supposes to store/fetch your models. I would just make a function `ensureHomeDir`.

Reply:
OK, I tried to refactor this into a function. Still unsure about error handling here.